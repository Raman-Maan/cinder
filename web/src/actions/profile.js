import * as types from './actionTypes';
import Auth from '../utils/authService';

export function profileErrored(bool) {
  return {
    type: types.PROFILE_ERROR,
    error: bool
  };
}

export function profileLoading(bool) {
  return {
    type: types.PROFILE_LOADING,
    loading: bool
  };
}

export function profileFetched(data) {
  return {
    type: types.PROFILE_FETCHED,
    profile: data
  };
}

export function fetchProfile() {
  return dispatch => {
    dispatch(profileLoading(true));

    fetch(`/api/users/${Auth.loggedInUser.id}/`)
      .then(res => {
        if (!res.ok) {
          throw Error(res.statusText);
        }
        dispatch(profileLoading(false));
        return res.json();
      })
      .then(data => dispatch(profileFetched(data)))
      .catch(() => dispatch(profileErrored(true)));
  };
}

export function showProfile() {
  return {
    type: types.SHOW_PROFILE
  };
}

export function hideProfile() {
  return {
    type: types.HIDE_PROFILE
  };
}
