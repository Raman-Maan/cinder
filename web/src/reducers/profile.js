import {
  PROFILE_ERROR,
  PROFILE_LOADING,
  PROFILE_FETCHED,
  SHOW_PROFILE,
  HIDE_PROFILE
} from '../actions/actionTypes';

const initState = {
  loading: false,
  errored: false,
  details: {},
  isVisible: false,
  message: ''
};

export function profile(state = initState, action) {
  switch (action.type) {
    case PROFILE_ERROR:
      return {
        ...state,
        loading: false,
        errored: true,
        message: action.payload
      };

    case PROFILE_LOADING:
      return {
        ...state,
        loading: true
      };

    case PROFILE_FETCHED:
      return {
        ...state,
        loading: false,
        details: action.payload
      };

    case SHOW_PROFILE:
      return {
        ...state,
        isVisible: true
      };

    case HIDE_PROFILE:
      return {
        ...state,
        isVisible: false
      };

    default:
      return state;
  }
}
