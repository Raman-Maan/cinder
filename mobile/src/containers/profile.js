import React from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Image } from 'react-native';
import { H1, View, Text, Container } from 'native-base';

import { fetchUserInfo } from '../actions';

import FilterElement from '../components/FilterElement';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
  },
  image: {
    height: 100,
    width: 100,
    marginTop: 20,
    borderRadius: 50,
    alignSelf: 'center',
  },
  filterContainer: {
    marginTop: 40,
  },
  userName: {
    marginTop: 10,
    alignSelf: 'center',
  },
  lineStyle: {
    alignSelf: 'stretch',
    borderWidth: 0.5,
    borderColor: 'grey',
    marginTop: 15,
    marginRight: 2,
    marginLeft: 2,
  },
});

export class Profile extends React.Component {
  static navigationOptions() {
    return {
      title: 'My Profile',
    };
  }

  constructor(props) {
    super(props);

    this.toggleModal = this.toggleModal.bind(this);
    this.onSelectChange = this.onSelectChange.bind(this);

    this.state = {
      gender: {
        modal: false,
        switch: false,
        values: [],
      },
    };
  }

  componentDidMount() {
    this.props.fetchUserInfo(3);
  }

  toggleModal(type) {
    this.setState(prev => ({
      [type]: {
        ...prev[type],
        modal: !prev[type].modal,
      },
    }));
  }

  onSelectChange(type) {
    return selections => this.setState(prev => ({
      [type]: {
        ...prev[type],
        values: selections,
      },
    }));
  }

  toggleSwitch(type) {
    this.setState(prev => ({
      [type]: {
        ...prev[type],
        switch: !prev[type].switch,
      },
    }));
  }

  render() {
    return (
      <Container style={styles.container}>
      <View>
        <Image accessibilityLabel={'profile_image'} style={styles.image} source={{ uri: this.props.userInfo.primaryPic }} />
        <H1 accessibilityLabel={'profile_name'} style={styles.userName}>{this.props.userInfo.userName}</H1>
        <View style={styles.lineStyle} />
      </View>

        <Container style={styles.filterContainer}>
          <Text>Filtering Options Here</Text>
          <FilterElement
            type="gender"
            switch={this.state.gender.switch}
            switchToggle={() => this.toggleSwitch('gender')}
            isVisible={this.state.gender.modal}
            onBackdropPress={() => this.toggleModal('gender')}
            selectedItems={this.state.gender.values}
            onSelectChange={this.onSelectChange('gender')}
          />
        </Container>
      </Container>
    );
  }
}

const mapStateToProps = state => ({
  userInfo: state.userInfo.userInfo,
});

const mapDispatchToProps = dispatch => ({
  fetchUserInfo: userID => dispatch(fetchUserInfo(userID)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
