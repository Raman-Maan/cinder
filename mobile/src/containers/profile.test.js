import React from 'react';
import renderer from 'react-test-renderer';

import { Profile } from './profile';

it('renders without crashing', () => {
  const rendered = renderer.create(<Profile />).toJSON();
  expect(rendered).toMatchSnapshot();
});
