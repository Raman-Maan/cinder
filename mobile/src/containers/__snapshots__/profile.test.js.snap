// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> fetchs data on load 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
  virtual={undefined}
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 50,
        "height": 100,
        "marginTop": 20,
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 27,
          "lineHeight": 32,
        },
        Object {
          "alignSelf": "center",
          "marginTop": 10,
        },
      ]
    }
    virtual={undefined}
  >
    Test Name
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "stretch",
          "borderColor": "grey",
          "borderWidth": 0.5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 15,
        },
      ]
    }
    virtual={undefined}
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "height": 1334,
        },
        Object {
          "marginTop": 40,
        },
      ]
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      Filtering Options Here
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#CCC",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
      virtual={undefined}
    >
      <RCTSwitch
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "marginBottom": -5,
              "marginTop": -5,
            },
          ]
        }
        value={false}
        virtual={undefined}
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#f4f4f4",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Filter by gender
        </Text>
      </View>
      <Modal
        animationType="none"
        fetchData={[Function]}
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onRequestClose={[Function]}
        onSelectChange={[Function]}
        refData={undefined}
        selectedItems={Array []}
        swipeThreshold={100}
        switch={false}
        switchToggle={[Function]}
        transparent={true}
        type="gender"
        visible={false}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
          testID={undefined}
        />
        <View
          keyboardVerticalOffset={0}
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Array [
                Object {
                  "margin": 37.5,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.8,
                },
                Object {
                  "margin": 0,
                },
              ],
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            fetchData={[Function]}
            hideModalContentWhileAnimating={false}
            onModalHide={[Function]}
            onSelectChange={[Function]}
            pointerEvents="box-none"
            refData={undefined}
            selectedItems={Array []}
            style={
              Object {
                "flex": 0.8,
                "justifyContent": "center",
                "margin": 37.5,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            swipeThreshold={100}
            switch={false}
            switchToggle={[Function]}
            type="gender"
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                  },
                ]
              }
              virtual={undefined}
            >
              <RCTScrollView
                dataSource={
                  ListViewDataSource {
                    "items": 0,
                  }
                }
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                style={Object {}}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;

exports[`<Profile> renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
  virtual={undefined}
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 50,
        "height": 100,
        "marginTop": 20,
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 27,
          "lineHeight": 32,
        },
        Object {
          "alignSelf": "center",
          "marginTop": 10,
        },
      ]
    }
    virtual={undefined}
  >
    Test Name
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "stretch",
          "borderColor": "grey",
          "borderWidth": 0.5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 15,
        },
      ]
    }
    virtual={undefined}
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "height": 1334,
        },
        Object {
          "marginTop": 40,
        },
      ]
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      Filtering Options Here
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#CCC",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
      virtual={undefined}
    >
      <RCTSwitch
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "marginBottom": -5,
              "marginTop": -5,
            },
          ]
        }
        value={false}
        virtual={undefined}
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#f4f4f4",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16.5,
              "lineHeight": 19,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Filter by gender
        </Text>
      </View>
      <Modal
        animationType="none"
        fetchData={[Function]}
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onRequestClose={[Function]}
        onSelectChange={[Function]}
        refData={undefined}
        selectedItems={Array []}
        swipeThreshold={100}
        switch={false}
        switchToggle={[Function]}
        transparent={true}
        type="gender"
        visible={false}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
          testID={undefined}
        />
        <View
          keyboardVerticalOffset={0}
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Array [
                Object {
                  "margin": 37.5,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.8,
                },
                Object {
                  "margin": 0,
                },
              ],
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            fetchData={[Function]}
            hideModalContentWhileAnimating={false}
            onModalHide={[Function]}
            onSelectChange={[Function]}
            pointerEvents="box-none"
            refData={undefined}
            selectedItems={Array []}
            style={
              Object {
                "flex": 0.8,
                "justifyContent": "center",
                "margin": 37.5,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
            swipeThreshold={100}
            switch={false}
            switchToggle={[Function]}
            type="gender"
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                  },
                ]
              }
              virtual={undefined}
            >
              <RCTScrollView
                dataSource={
                  ListViewDataSource {
                    "items": 0,
                  }
                }
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                style={Object {}}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
